backend/routes.js

```diff
@@ -1,53 +1,514 @@
+            });
+        });
+
+        /* Persistent Datastores */
+
+        // return the list of registered queries
+        server.get('/queries', function getQueriesHandler(req, res, next) {
+            res.json(200, getQueries());
+        });
+
+        server.get('/queries/:fid', function getQueriesFidHandler(req, res, next) {
+            const query = getQuery(req.params.fid);
+            if (query) {
+                res.json(200, query);
+            } else {
+                res.json(404, {});
+            }
+        });
+
+        // register or overwrite a query
+        /*
+         * TODO - Updating a query should be a PATCH or PUT under
+         * the endpoint `/queries/:fid`
+         */
+        server.post('/queries', function postQueriesHandler(req, res, next) {
+            // Make the query and update the user's grid
+            const {fid, uids, query, connectionId, requestor} = req.params;
 
-    server.get('/status', serveHttpsStatus);
+            // Check that the user has permission to edit the grid
 
-    // setup v1 endpoints
-    ENDPOINTS.v1.forEach (endpoint => {
-        server.get(`/v1/${endpoint}`, serverRespondMethod);
-        server.post(`/v1/${endpoint}`, serverRespondMethod);
-    });
+            checkWritePermissions(fid, requestor)
+            .then(function nowQueryAndUpdateGrid() {
+                return that.queryScheduler.queryAndUpdateGrid(
+                    fid, uids, query, connectionId, requestor
+                )
+            })
+            .then(function returnSuccess() {
+                let status;
+                if (getQuery(req.params.fid)) {
+                    // TODO - Technically, this should be
+                    // under the endpoint `/queries/:fid`
+                    status = 200;
+                } else {
+                    status = 201;
+                }
+                that.queryScheduler.scheduleQuery(req.params);
+                res.json(status, {});
+            })
+            .catch(function returnError(error) {
+                Logger.log(error, 0);
+                res.json(400, {error: {message: error.message}});
+            });
 
+        });
+
+        // TODO - This endpoint is untested
+        server.get('has-certs', (req, res, next) => {
+            if (isEmpty(getCerts())) {
+                res.json(404, false);
+            } else {
+                res.json(200, true);
+            }
+        });
+
+        // TODO - This endpoint is untested
+        server.get('is-url-redirected', (req, res, next) => {
+            const contents = fs.readFileSync(HOSTS);
+            if (contents.indexOf(getSetting('CONNECTOR_HTTPS_DOMAIN')) > -1) {
+                res.json(200, true);
+            } else {
+                res.json(404, false);
+            }
+        });
+
+        // TODO - This endpoint is untested
+        server.get('start-temp-https-server', (req, res, next) => {
```

Comment:
since these 3 endpoints are only used for internal use, maybe we can prefix them with something like `/internal/has-certs`? that way it's clear that we are allowed to change them in the future - no other services depend on them